### PR TITLE
Fixed #32106 -- Preserved HTTP_HOST in test Client when following redirects.

### PR DESCRIPTION
**Pull request by bcail**

### CODE REVIEW

File: tests/test_client/tests.py [hunk 1]
```diff
@@ -856,6 +856,36 @@ def test_external_redirect(self):
...
+            "SERVER_PORT": "80",
+            "wsgi.url_scheme": "https",
+            "QUERY_STRING": "",
+            "SERVER_NAME": "www.djangoproject.com",
+            "HTTP_HOST": "www.djangoproject.com",
+        }
+        calls = m.mock_calls
+
+        self.assertEqual(calls[0], mock.call(**call_1_params))
+        self.assertEqual(calls[1], mock.call(**call_2_params))
```
Comment (felixxm) [inline tests/test_client/tests.py:887]: There is a lot of mocking in this test :disappointed: Is it not testable with normal flow? If not, there is probably nothing to fix. Have you tried adding a test for the scenario described in the ticket?

Comment (bcail) [reply tests/test_client/tests.py:887]: Thanks, @felixxm. One challenge with the ticket scenario is that it's talking about a redirect to a different hostname. Mocking here lets me have a redirect to a different hostname, and verify that the test client is setting HTTP_HOST for the second request. Is there a way to test requests to two different hostnames without mocking?

Comment (shangxiao) [reply tests/test_client/tests.py:887]: We can totally test this without mocking 👍
See PR with failing test:  https://github.com/django/django/pull/17430

With Carlton's patch, the test passes.

Comment (bcail) [reply tests/test_client/tests.py:887]: Great, thanks @shangxiao. I'll close this PR and you can use yours.

**Comment by bcail** [comment]
Closing in favor of #17430.

File: tests/test_client/tests.py (hunk 1 above)
Comment (shangxiao) [reply tests/test_client/tests.py:887]: Oh my intention was to demonstrate a test alternative and you could use it as a guide... (If we agree testing that way was beneficial).  I mean, please feel free to continue, you deserve the credit for taking the initiative on this👍

Comment (bcail) [reply tests/test_client/tests.py:887]: OK, either way is fine with me. I'll update this PR based on your PR then.

**Comment by bcail** [comment]
@felixxm I updated this PR based on @shangxiao's PR.

**Comment by felixxm** [comment]
@bcail Thanks :+1: I couldn't push tiny edits to this PR so I opened #17431.

**Comment by shangxiao** [comment]
@bcail just fyi be sure to check "Allow edits by maintainers" when submitting a PR. Hopefully there's another ticket you can pick up soon if you're looking to help contribute, I'll gladly help in any way 👍

**Comment by bcail** [comment]
@felixxm @shangxiao For the "Allow edits by maintainers" option - unfortunately, I'm not able to do that, see https://github.com/orgs/community/discussions/5634. Hopefully they'll fix that soon.

**Pull request closed without merging**